tests_metricflow/snapshots/test_custom_granularity.py/SqlQueryPlan/DuckDB/test_conversion_metric_with_custom_granularity__plan0_optimized.sql

```diff
@@ -0,0 +1,139 @@
+-- Compute Metrics via Expressions
+SELECT
+  metric_time__martian_day
+  , CAST(buys AS DOUBLE) / CAST(NULLIF(visits, 0) AS DOUBLE) AS visit_buy_conversion_rate_7days
+FROM (
+  -- Combine Aggregated Outputs
+  SELECT
+    COALESCE(subq_24.metric_time__martian_day, subq_36.metric_time__martian_day) AS metric_time__martian_day
+    , MAX(subq_24.visits) AS visits
+    , MAX(subq_36.buys) AS buys
+  FROM (
+    -- Pass Only Elements: ['visits', 'metric_time__day']
+    -- Join to Custom Granularity Dataset
+    -- Pass Only Elements: ['visits', 'metric_time__martian_day']
+    -- Aggregate Measures
+    SELECT
+      subq_21.martian_day AS metric_time__martian_day
+      , SUM(subq_20.visits) AS visits
+    FROM (
+      -- Read Elements From Semantic Model 'visits_source'
+      -- Metric Time Dimension 'ds'
+      SELECT
+        DATE_TRUNC('day', ds) AS metric_time__day
+        , 1 AS visits
+      FROM ***************************.fct_visits visits_source_src_28000
+    ) subq_20
+    LEFT OUTER JOIN
+      ***************************.mf_time_spine subq_21
+    ON
+      subq_20.metric_time__day = subq_21.ds
+    GROUP BY
+      subq_21.martian_day
+  ) subq_24
+  FULL OUTER JOIN (
+    -- Find conversions for user within the range of 7 day
+    -- Pass Only Elements: ['buys', 'metric_time__martian_day']
+    -- Pass Only Elements: ['buys', 'metric_time__martian_day']
+    -- Aggregate Measures
+    SELECT
+      metric_time__martian_day
+      , SUM(buys) AS buys
+    FROM (
+      -- Dedupe the fanout with mf_internal_uuid in the conversion data set
+      SELECT DISTINCT
+        FIRST_VALUE(subq_28.visits) OVER (
+          PARTITION BY
+            subq_31.user
+            , subq_31.ds__day
+            , subq_31.mf_internal_uuid
+          ORDER BY subq_28.ds__day DESC
+          ROWS BETWEEN UNBOUNDED PRECEDING AND UNBOUNDED FOLLOWING
+        ) AS visits
+        , FIRST_VALUE(subq_28.metric_time__martian_day) OVER (
+          PARTITION BY
+            subq_31.user
+            , subq_31.ds__day
+            , subq_31.mf_internal_uuid
+          ORDER BY subq_28.ds__day DESC
+          ROWS BETWEEN UNBOUNDED PRECEDING AND UNBOUNDED FOLLOWING
+        ) AS metric_time__martian_day
+        , FIRST_VALUE(subq_28.ds__day) OVER (
+          PARTITION BY
+            subq_31.user
+            , subq_31.ds__day
+            , subq_31.mf_internal_uuid
+          ORDER BY subq_28.ds__day DESC
+          ROWS BETWEEN UNBOUNDED PRECEDING AND UNBOUNDED FOLLOWING
+        ) AS ds__day
+        , FIRST_VALUE(subq_28.metric_time__day) OVER (
```

Comment:
Is it expected that we still use `metric_time__day` in the interim calculations here? Looks like we use both `metric_time__day` and `metric_time__martian_day` in the window functions, even though only `metric_time__martian_day` was in the group by.

Reply:
how does cumulative windows work for custom grains? We have `metric_time__day` here because that's what's used for the conversion window calculation (ie., look at line 126 below for the join condition). So `metric_time__day` is here because it's used for the conversion window and `metric_time__martian_day` is there because we will need it when grouping by it.

Reply:
actually, looking at this, we use `ds__day` to do the calculation. There's probably some wonky thing happening with the column's I passed into the conversion node, not blocking, but I'll take a look.

Reply:
Cumulative windows can't use custom grains yet - that's part of the implementation we were discussing at MF team meeting today!
I think non-default grains are relevant, though. If you look at [this SQL snapshot](https://github.com/dbt-labs/metricflow/blob/fc2be4c258128ffee8fd3f4fa85382e52a2dad79/tests_metricflow/snapshots/test_cumulative_metric_rendering.py/SqlQueryPlan/Trino/test_window_metric_with_non_default_grain__plan0_optimized.sql#L10-L11) you can see an example of the window function we implement for that.

Reply:
But it sounds like the PR you put up today was a fix for what you mentioned here!